compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1607,15 +1611,15 @@ impl<'tcx> InferCtxt<'tcx> {
                 // If `inlined_probe_value` returns a value it's always a
                 // `ty::Int(_)` or `ty::UInt(_)`, which never matches a
                 // `ty::Infer(_)`.
-                self.inner.borrow_mut().int_unification_table().inlined_probe_value(v).is_some()
+                !self.inner.borrow_mut().int_unification_table().inlined_probe_value(v).is_unknown()
```

Comment:
```suggestion
                self.inner.borrow_mut().int_unification_table().inlined_probe_value(v).is_known()
```
double negation is confusing esp given the length of the negated condition